pulsar-client/src/main/java/org/apache/pulsar/client/api/Authentication.java

```diff
@@ -40,13 +53,46 @@
     AuthenticationDataProvider getAuthData() throws PulsarClientException;
 
     /**
-     * Configure the authentication plugins with the supplied parameters
+     * Configure the authentication plugins with the Map parameter
      *
      * @param authParams
      */
     void configure(Map<String, String> authParams);
 
     /**
+     * Configure the authentication plugins with the String parameter
+     *
+     * @param authParams
+     */
+    default void configure(String authParams) {
+        Map<String, String> authParamsMap = new HashMap<>();
+        Logger LOG = LoggerFactory.getLogger(Authentication.class);
```

Comment:
logger should be declared as `private static final` at the class scope

Reply:
`private static final` can not be declared at  the class scope because Authentication.java is an interface.
https://stackoverflow.com/questions/31452118/why-private-static-field-is-not-allowed-in-java-8-interface

Reply:
Ok, though instead of having the parsing logic here in the interface, we could have it in a different place and then call it. eg: 

```java
default void configure(String authParamsStr) {
   Map<String, String> authParams = AuthenticationUtil.configureFromJsonString(authParamsStr);
   configure(authParams);
}
```

Reply:
Do we really want to have the two lines here? If we move parsing logic to somewhere else, we can deprecate the old interface and will have this only one universal configure interface eventually, which is more beautiful API I think.

Reply:
I guess the intention for using `default` method was to avoid breaking the `Authentication` API. I guess we could as well break it right now, since I'm not aware of any 3rd party Authentication implementations at this point.

Reply:
I guess so, but probably we can achieve the both, keeping compatibility and having one configure (and deprecated old one) if we introduce an interface I wrote on the issue.